pycbc/io/hdf.py

```diff
@@ -822,14 +870,18 @@ def to_coinc_xml_object(self, file_name):
             # FIXME: As two-ifo is hardcoded loop over all ifos
             sngl_combined_mchirp = 0
             sngl_combined_mtot = 0
+            net_snrsq = 0
             for ifo in ifos:
-                sngl_id = self.trig_id[ifo][idx]
-                event_id = lsctables.SnglInspiralID(sngl_id)
+                if not sngl_col_vals['snr'][ifo][1][idx]:
+                    continue
+                event_id = lsctables.SnglInspiralID(sngl_event_count)
```

Comment:
So here I'm actually fixing a pre-existing edge-case bug.

Before, sngl_id was chosen as the location of the trigger in the corresponding HDF_TRIGGER_MERGE file. Now I'm choosing incremental integers. The *only* requirement for the sngl_id is that it is unique. (It's used to map between sngls and coincidences in the XML format). In the old code it was *possible* to get non-unique entries ... which would lead to the sort of problem Tito reported in #3263 (although surely this cannot be the same thing).

.. I can't really put a description of the XML format in the in-line comments here. So I guess I have to assume some familiarity in this block, otherwise the whole thing just makes no sense at all! ... I also don't know a good place to point to for documentation.

Reply:
okay, if the point is 'give it a unique number' then this is okay